test/test_serializers/test_serializer_longturtle_sort.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python3
+
+# Portions of this file contributed by NIST are governed by the
+# following statement:
+#
+# This software was developed at the National Institute of Standards
+# and Technology by employees of the Federal Government in the course
+# of their official duties. Pursuant to Title 17 Section 105 of the
+# United States Code, this software is not subject to copyright
+# protection within the United States. NIST assumes no responsibility
+# whatsoever for its use by other parties, and makes no guarantees,
+# expressed or implied, about its quality, reliability, or any other
+# characteristic.
+#
+# We would appreciate acknowledgement if the software is used.
+
+import random
+from collections import defaultdict
+
+from rdflib import RDFS, BNode, Graph, Literal, Namespace, URIRef
+
+EX = Namespace("http://example.org/ex/")
+
+
+def test_sort_semiblank_graph() -> None:
+    """
+    This test reviews whether the output of the Turtle form is
+    consistent when involving repeated generates with blank nodes.
+    """
+
+    serialization_counter: defaultdict[str, int] = defaultdict(int)
+
+    first_graph_text: str = ""
+
+    # Use a fixed sequence of once-but-no-longer random values for more
+    # consistent test results.
+    nonrandom_shuffler = random.Random(1234)
+    for x in range(1, 10):
+        graph = Graph()
+        graph.bind("ex", EX)
+        graph.bind("rdfs", RDFS)
+
+        graph.add((EX.A, RDFS.comment, Literal("Thing A")))
+        graph.add((EX.B, RDFS.comment, Literal("Thing B")))
+        graph.add((EX.C, RDFS.comment, Literal("Thing C")))
+
+        nodes: list[URIRef] = [EX.A, EX.B, EX.C, EX.B]
+        nonrandom_shuffler.shuffle(nodes)
+        for node in nodes:
+            # Instantiate one bnode per URIRef node.
+            graph.add((BNode(), RDFS.seeAlso, node))
+
+        nesteds: list[URIRef] = [EX.A, EX.B, EX.C]
+        nonrandom_shuffler.shuffle(nesteds)
+        for nested in nesteds:
+            # Instantiate a nested node reference.
+            outer_node = BNode()
+            inner_node = BNode()
+            graph.add((outer_node, EX.has, inner_node))
+            graph.add((inner_node, RDFS.seeAlso, nested))
+
+        graph_text = graph.serialize(format="longturtle", sort=True)
+        if first_graph_text == "":
+            first_graph_text = graph_text
+
+        serialization_counter[graph_text] += 1
+
+    expected_serialization = """\
+PREFIX ns1: <http://example.org/ex/>
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+
+ns1:A
+    rdfs:comment "Thing A" ;
+.
+
+ns1:C
+    rdfs:comment "Thing C" ;
+.
+
+ns1:B
+    rdfs:comment "Thing B" ;
+.
+
+[]    ns1:has
+        [
+            rdfs:seeAlso ns1:A ;
+        ] ; ;
+.
+
+[]    rdfs:seeAlso ns1:B ; ;
```

Comment:
Two semicolons?

Reply:
That's strange. I've never seen that before. I just tried serialising the same data with rdflib v7.0.0 and 7.1.1, and it's producing the same result.

Reply:
The extra semicolons were introduced in this commit https://github.com/RDFLib/rdflib/pull/2500/commits/ba585bd820c2ac6610d8a93026065b602a7a1910. It appears that they were added to ensure that every statement in an object ended with a semicolon. But because the `s_squared ` method was already writing a semicolon, it would double up for subjects that are blank nodes.

https://github.com/RDFLib/rdflib/blob/e8f61d4b5019040f097e560c3c1c8ecb41dd3fac/rdflib/plugins/serializers/longturtle.py#L184-L190

Fixed in https://github.com/RDFLib/rdflib/pull/3008/commits/412fb5d63d79c5d5816e05751e6be6bd791b1385.